content/posts/0.27-release.md

```diff
@@ -0,0 +1,166 @@
+an error message that we are planning to throw away.
+
+The second improvement we made was to optimize away temporary bound methods.
+Python code like
+```py
+a.f()
+```
+means first look up `a.f` and then call the result, which is translated into
+bytecode like:
+```
+LOAD a
+LOAD_ATTR f
+CALL
+```
+The function `a.f` receives `a` as the first `self` argument, so it needs to be
+a special bound method object that knows about the `self` argument. However, we
+only use this object once to call it and then we are going to throw it away. We
+can avoid allocating and destroying an object when we call a method by calling
+`type(a).f(a)`. The `LOAD_ATTR` opcode has a special argument that indicates
+that the next opcode is going to be `CALL` and in that case calls a method named
+`_PyObject_GetMethod` instead of the typical `PyObject_GetAttr` to perform the
+attribute lookup. We patched `_PyObject_GetMethod` to have special handling for
+`JsProxy` objects so that we could optimize away the temporary for JS objects too.
+
+### A potential future performance improvement
+
+In the case of lookup on a `JsProxy`, there are other bytecode patterns where we
+could optimize away temporaries. For instance, if `a` is a `JsProxy` for the
+JavaScript object `{ x: { y: 2 } }` then `a.x.y` creates a temporary `JsProxy`
+pointing to `{ y: 2 }` and then looks up `y` on the temporary and returns `2`.
+The bytecode looks like:
+```
+LOAD a
+LOAD_ATTR x
+LOAD_ATTR y
+```
+If we know that we were going to use the result of the first `LOAD_ATTR x`
+immediately, we could avoid creating the temporary. Unfortunately, unlike the
+method call example, native Python code doesn't create a temporary here so
+optimizing this would involve more invasive patches to the interpreter than
+removing the temporary bound methods.
+
+## Work in Progress / Roadmap
+
+TODO
+
+## Acknowledgements
+
+Thanks to Agriya Khetarpal, Loïc Estève, and Ralf Gommers for their work helping ensure scientific Python packages are well supported in Pyodide.
+Thanks to Joe Marshall for his huge contribution on PyArrow Support.
+
+Additionally, we always appreciate the support and assistance from the Emscripten team.
+
+The following people commited to Pyodide in this release:
+
+Agriya Khetarpal, airen1986, Andrei V. Plamada, Andrew Moon, Bart Broere, Carlo 
+Piovesan, Castedo Ellerman, Chris Pyles, Christian Clauss, chrysn, Deepak 
+Cherian, Eli Lamb, Em Zhan, Eric Brown, Gyeongjae Choi, Hanno Rein, Henry 
+Schreiner, Hood Chatham, Ian Thomas, James J Balamuta, JHM Darbyshire, Jiefu7, 
+Joel Ostblom, Joe Marshall, josephrocca, Juniper Tyree, Kellen Malek, Kyle 
+Barron, Loïc Estève, Luiz Irber, Maarten Breddels, Marco Edward Gorelli, 
+Marianne Corvellec, M Bussonnier, Muspi Merol, Myles Scolnick, Nick Altmann, 
+Olivier Grisel, Oscar Benjamin, Péter Gyarmati, Phillip Cloud, pyodide-pr-bot, 
+Riya Sinha, swnf, Szabolcs Dombi, Victor Blomqvist, Yan Wong, YISH, Zsolt
```

Comment:
Here is an updated list from earlier today:

```suggestion
The following people committed to Pyodide in this release:

Agriya Khetarpal, Andrei V. Plamada, Andrew Moon, Bart Broere, Carlo
Piovesan, Castedo Ellerman, Chris Pyles, Christian Clauss, Deepak Cherian,
Eli Lamb, Em Zhan, Eric Brown, Gyeongjae Choi, Hanno Rein, Henry Schreiner,
Hood Chatham, Ian Thomas, JHM Darbyshire, James J Balamuta,
Jiefu7, Joe Marshall, Joel Ostblom, Juniper Tyree, Kellen Malek, Kyle Barron,
Loïc Estève, Luiz Irber, M Bussonnier, Maarten Breddels, Marco Edward Gorelli,
Marianne Corvellec, Muspi Merol, Myles Scolnick, Nick Altmann, Olivier Grisel,
Oscar Benjamin, Péter Gyarmati, Phillip Cloud, Riya Sinha, Szabolcs Dombi,
Victor Blomqvist, YISH, Yan Wong, Zsolt Dollenstein, airen1986, chrysn,
josephrocca, swnf
```

I used this command to calculate and sort it:

```bash
git log --pretty=format:"%an" 0.26.0..HEAD \
| sort -u | grep -v "\[bot\]\|pyodide-pr-bot" | \
paste -sd "," - | sed 's/,/, /g'
```